plugins/pulp_rpm/plugins/migrations/0028_standard_storage_path.py

```diff
@@ -0,0 +1,124 @@
+from pulp.server.db import connection
+
+from pulp.plugins.migration.standard_storage_path import Migration, Plan
+
+
+def migrate(*args, **kwargs):
+    """
+    Migrate content units to use the standard storage path introduced in pulp 2.8.
+    """
+    migration = Migration()
+    migration.add(RPM())
+    migration.add(SRPM())
+    migration.add(DRPM())
+    migration.add(Distribution())
+    migration.add(YumMetadata())
+    migration.add(ISO())
+    migration()
+
+
+class Package(Plan):
+    """
+    The (base) migration plan for package units.
+    """
+
+    key_fields = (
```

Comment:
Yeah.  I was on the fence on this and never intended for the plugins to override `Plan`.  I originally wrote the plugins to just have a factory function for each type that created and returned a configured `Plan`.  After seeing how this plays out and especially given what you pointed out about the `key_fields` I'm in favor going back to that approach.  Like:
```
def rpm_plan():
  return Plan(connection.get_collection('units_rpm'), ('name', 'version', ..))
```
then do:
```
migration.add(rpm_plan())
```
Could to it all in the `migrate()` function but might be more testable separated out.  Or, maybe not.